app/graphql/mutations/profile_link/create.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Mutations::ProfileLink::Create < Mutations::Base
+  include FancyMutation
+
+  description 'Create a profile link'
+
+  input do
+    argument :url, String,
+      required: true,
+      description: 'The url of the profile link',
+      as: :profile_link_url
+    argument :profile_link_site,
+      Types::Enum::ProfileLinksSites,
+      required: true,
+      description: 'The website.'
+  end
+  result Types::SiteLink
+  errors Types::Errors::NotAuthenticated,
+    Types::Errors::NotAuthorized,
+    Types::Errors::NotFound,
+    Types::Errors::Validation
+
+  def ready?(**)
+    authenticate!
+    return errors << Types::Errors::NotAuthenticated.build if current_user.nil?
+    profile_link = ProfileLink.find_by(
+      profile_link_site_id: profile_link,
+      user_id: current_user.id
+    )
+    raise GraphQL::ExecutionError, 'You already have this profile link.' if profile_link
```

Comment:
A better way to handle this is probably to add a `validates :profile_link_site, uniqueness: { scope: :user }` on the ProfileLink model, and rely on `Types::Errors::Validation` to handle this